c_common/models/data_speed_up_packet_gatherer/data_speed_up_packet_gatherer.c

```diff
@@ -55,33 +146,382 @@ static uint32_t seq_num = FIRST_SEQ_NUM;
+       ITEMS_PER_DATA_INDEX - COMMAND_ID_SIZE_IN_ELEMENTS -
+       TOTAL_MISSING_SEQ_PACKETS_IN_ELEMENTS);
+   return ceil(missing_seq / (
+       ITEMS_PER_DATA_INDEX - COMMAND_ID_SIZE_IN_ELEMENTS));
 }
 
+//! updates the memory in msg, and then sends missing seq if full. updates
+//! command code accordingly.
+//! \param[in] missing_seq_num: the missing seq num to fill in
+//! \param[in] position_in_data: where in msg we are.
+uint update_and_send_sdp_if_required(
+        uint missing_seq_num, uint position_in_data){
+    //log_info("adding missing seq num %d\n", missing_seq_num);
+    my_msg.data[position_in_data] = missing_seq_num;
+    position_in_data ++;
+    if (position_in_data == ITEMS_PER_DATA_INDEX){
+        //log_info("sending missing data packet");
+        my_msg.length = LENGTH_OF_SDP_HEADER + (
+            position_in_data * WORD_TO_BYTE_MULTIPLIER);
+        while (!spin1_send_sdp_msg((sdp_msg_t *) &my_msg, SDP_TIMEOUT)) {
+	        spin1_delay_us(MESSAGE_DELAY_TIME_WHEN_FAIL);
+        }
+        my_msg.data[COMMAND_ID_POSITION] =
+            SDP_PACKET_SEND_MISSING_SEQ_DATA_IN_COMMAND_ID;
+        position_in_data = 1;
+    }
+    return position_in_data;
+}
+
+//! \brief calculates the new sdram location for a given seq num
+//! \param[in] seq_num: the seq num to figure offset for
+//! \return the new sdram location.
+uint calculate_sdram_address_from_seq_num(uint seq_num){
+    //log_info("seq num is %d", seq_num);
+    if (seq_num == 0){
+        //log_info("first packet");
+        return start_sdram_address;
+    }
+    if (seq_num == 1){
+        //log_info("first data packet");
+        return start_sdram_address + (
+            DATA_IN_FULL_PACKET_WITH_ADDRESS_NUM * WORD_TO_BYTE_MULTIPLIER);
+    }
+    else{
+        //log_info("start address is %d, first packet size is %d, seq num offset is %d",
+        //    start_sdram_address, DATA_IN_FULL_PACKET_WITH_ADDRESS_NUM * WORD_TO_BYTE_MULTIPLIER,
+        //    WORD_TO_BYTE_MULTIPLIER * DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM * seq_num);
+        //log_info("part 1 =%d, part 2 =%d part 3 =%d part 4= %d, part5 = %d",
+        //         WORD_TO_BYTE_MULTIPLIER, DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM, seq_num,
+        //         WORD_TO_BYTE_MULTIPLIER * DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM,
+        //         DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM * seq_num);
+        //log_info("issue %d", value);
+        return start_sdram_address + (
+            (DATA_IN_FULL_PACKET_WITH_ADDRESS_NUM * WORD_TO_BYTE_MULTIPLIER) +
+            (WORD_TO_BYTE_MULTIPLIER *
+             DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM * seq_num));
+    }
+}
+
+//! \brief searches through received seq nums and transmits missing ones back
+//! to host for retransmission
+void process_missing_seq_nums_and_request_retransmission(){
+
+    // check that missing seq transmission is actually needed, or
+    // have we finished
+    log_info(" total recieved = %d, max seq is %d\n",
```

Comment:
Do we really want to have messages being written to the iobufs?

Reply:
was left over debug messages, have commented out.